File: todo-server/src/main/java/com/example/todoserver/models/ToDo.java
package com.example.todoserver.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "todo")
public class ToDo {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String text;
    private boolean checked;
}

File: todo-android/app/src/main/java/com/example/todo/models/ToDo.kt
package com.example.todo.models

import com.squareup.moshi.Json

data class ToDo(
    @Json(name = "id")
    var id: Long?,
    @Json(name = "text")
    var text: String?,
    @Json(name = "checked")
    var checked: Boolean?
)